In [1]:
library(annotinder)

In [48]:
# final tweets
df <- read.csv("../data/final_stratified_immigration_non_immigration.csv")

In [50]:
### Maak column aan met instructies voor coderen in conditie 3, 4, 5 (author = zichtbaar)

In [51]:
df$id <- as.character(df$tweet_id)
# Print the data frame
# Combine author and twitter_handle columns with Markdown formatting for author name
df$author_with_handle <- paste0("**", df$Name, "** ", df$twitter_username, "")
# Create the new column with party name bolded using Markdown
df$review_message <- paste0(
  "Beoordeel het volgende social media bericht afkomstig van **", 
  df$normalized_party, "** lid ", df$author , ":"
)
df

X    Unnamed..0_x tweet_id     timestamp          
1     0   362122       1.250008e+18 2020-04-14 08:27:49
2     1    16705       1.659506e+18 2023-05-19 10:25:33
3     2   509211       1.497219e+18 2022-02-25 13:36:58
4     3   907773       1.421811e+18 2021-08-01 10:34:06
5     4   124597       1.371887e+18 2021-03-16 17:11:48
6     5   809894       1.347815e+18 2021-01-09 06:59:13
7     6   735111       1.510889e+18 2022-04-04 07:55:08
8     7   550884       1.235944e+18 2020-03-06 15:01:54
9     8   880904       1.595397e+18 2022-11-23 12:42:17
10    9   187249       1.370694e+18 2021-03-13 10:13:10
11   10   421438       1.432387e+18 2021-08-30 14:57:08
12   11   462720       1.274961e+18 2020-06-22 05:03:04
13   12   647109       1.623974e+18 2023-02-10 09:16:32
14   13   480823       1.430649e+18 2021-08-25 19:49:24
15   14   476606       1.626859e+18 2023-02-18 08:21:19
16   15    86541       1.414568e+18 2021-07-12 10:51:53
17   16    80523       1.371540e+18 2021-03-15 18:14:16
18   17   621165       1.567817e+18 2022-09-08 08:06:19
19   18   108330       1.312074e+18 2020-10-02 14:55:36
20   19   288531       1.340701e+18 2020-12-20 15:49:07
21   20   234217       1.498349e+18 2022-02-28 16:25:14
22   21   510869       1.319612e+18 2020-10-23 12:10:35
23   22    90424       1.665245e+18 2023-06-04 06:32:50
24   23    18105       1.505988e+18 2022-03-21 18:23:22
25   24   849537       1.460151e+18 2021-11-15 06:39:44
26   25   653691       1.671541e+18 2023-06-21 15:31:52
27   26   881370       1.224724e+18 2020-02-04 15:58:32
28   27   610902       1.504135e+18 2022-03-16 16:38:24
29   28   837317       1.659857e+18 2023-05-20 09:40:47
30   29   307772       1.457807e+18 2021-11-08 19:26:10
⋮    ⋮    ⋮            ⋮            ⋮                  
3301 3300 456762       1.449462e+18 2021-10-16 17:48:45
3302 3301 268937       1.229024e+18 2020-02-16 12:43:37
3303 3302 606525       1.564705e+18 2022-08-30 18:01:36
3304 3303 115563       1.387130e+18 2021-04-27 19:42:21
3305 3304 426402       1.380388e+18 2021-04-09 03:13:13
3306 3305 569392       1.273259e+18 2020-06-17 12:18:38
3307 3306  86810       1.520674e+18 2022-05-01 07:57:28
3308 3307 919387       1.237136e+18 2020-03-09 21:01:02
3309 3308  29746       1.451476e+18 2021-10-22 07:09:39
3310 3309 707773       1.635647e+18 2023-03-14 14:20:28
3311 3310 867977       1.354805e+18 2021-01-28 13:53:50
3312 3311 407489       1.482322e+18 2022-01-15 11:02:29
3313 3312 777093       1.399134e+18 2021-05-30 20:41:01
3314 3313 736138       1.418810e+18 2021-07-24 05:46:04
3315 3314 714773       1.428377e+18 2021-08-19 13:22:46
3316 3315 366844       1.620690e+18 2023-02-01 07:47:18
3317 3316  73039       1.322574e+18 2020-10-31 15:21:21
3318 3317 396767       1.421465e+18 2021-07-31 11:35:41
3319 3318 332400       1.719469e+18 2023-10-31 21:40:13
3320 3319 424519       1.270323e+18 2020-06-09 09:53:32
3321 3320 220077       1.553661e+18 2022-07-31 06:38:45
3322 3321  30988       1.339151e+18 2020-12-16 09:12:35
3323 3322 227780       1.710031e+18 2023-10-05 20:34:54
3324 3323 255816       1.500923e+18 2022-03-07 18:55:50
3325 3324 738688       1.258747e+18 2020-05-08 11:14:05
3326 3325 810865       1.369406e+18 2021-03-09 20:54:45
3327 3326 739612       1.223883e+18 2020-02-02 08:18:06
3328 3327 229422       1.257942e+18 2020-05-06 05:56:04
3329 3328 838411       1.632700e+18 2023-03-06 11:08:57
3330 3329 660660       1.485550e+18 2022-01-24 09:50:04
     text                                                                                                                                                                                                                                                                                                                   
1    @jpaternotte @GoosR @WNLVandaag Net genoeg tijd om na te denken over een repliek! ;-)                                                                                                                              

In [4]:
read_and_concatenate <- function(file_path) {
    instructies <- readLines(file_path, warn = FALSE)
    text_string <- paste(instructies, collapse = "\n")
    return(text_string)
}

### Part 1:  Training

<div style="background-color:yellow; padding:10px;">
@Kasper: De training moet uiteraard voor de final coding job komen. Zo zorg ik ervoor dat dat het final codebook + de final units na de training komen?
</div>


In [39]:
## Codebook annotation

## misinformation
political_stance_training <- question("political_stance_training", 
                  "In welke mate reflecteert dit bericht een links- of rechts politiek standpunt?",
             instruction = read_and_concatenate("instructions/political_stance.txt")  , 
             codes = c( darkgrey = 'Volledig links/ progressief -- Links/ progressief -- Enigszins links/progressief',
                        grey = 'Neutraal/ onduidelijk', 
                        darkgrey = 'Engiszins rechts/ conservatief -- Rechts/conservatief -- Helemaal rechts/ conservatief'))

## sentiment
sentiment_training <- question("sentiment_training", 
                      "Is deze tweet negatief of positief van toon?",
             instruction = read_and_concatenate("instructions/sentiment.txt") , 
             codes = c( darkred = 'Zeer negatief -- Negatief -- Enigszins negatief',
                        grey = 'Neutral/onduidelijk', 
                        darkgreen = 'Enigszins positief -- Positief -- Zeer positief'))

## toxic
toxic_training <- question("toxic_training", 
                  "Bevat deze tweet toxische taal?",
             instruction = read_and_concatenate("instructions/toxic.txt")  , 
             codes = c( darkgreen = 'Zeer beleefd/ vriendelijk -- Beleefd/ vriendelijk -- Enigzins beleefd/vriendelijk',
                        grey = 'Neutraal/ onduidelijk', 
                        darkred = 'Enigzins toxisch -- Toxisch -- Zeer toxisch'))

## misinformation
misinformation_training  <- question("misinformation_training", "Bevat deze tweet misinformatie?",
             instruction = read_and_concatenate("instructions/misinformation.txt")  , 
             codes = c( darkgreen = 'Geen/ Waarschijnlijk geen/ Mogelijk geen misinformatie',
                        grey = 'Neutraal/ onduidelijk', 
                        darkred = 'Mogelijk/ Waarschijnlijk/ Zeker misinformatie'))




intro_unit_1 = create_unit("welkom", 
                         set_markdown("intro",  read_and_concatenate('instructions/general_instructions.txt'), text_size = 1.5, align = 'center'), 
                         set_markdown("general_definitions", read_and_concatenate('instructions/general_definitions.txt'), text_size = 1.2, align = 'center', background = 'lightgrey', marginTop = '3rem'),
                         set_question('confirm', question = 'Ga door als u klaar bent met lezen', type='confirm'))


intro_unit_2 = create_unit("welkom2", 
                         set_markdown("intro",  
                                      "Voordat het volgende deel begint, oefenen we met een aantal Tweets.\nBeoordeel alstublieft de volgende Tweets:"),
                                         set_question('confirm', question = 'Ga door als u klaar bent met lezen', type='confirm')) 



train_unit_1 <- create_unit(
  'test vraag 1',
  type = 'train',
  set_text('text', 'We hebben NU een rechts kabinet nodig! Geen GL/PvdA! Stop die asielwaanzin en bouw meer woningen! Nederland kan niet langer naar die linkse flauwekul luisteren!'),
  set_train(variable = "sentiment_training", 
            value = 'Zeer negatief -- Negatief -- Enigszins negatief', 
            message = "Het sentiment van de tweet is negatief vanwege de negatieve en verontwaardigde toon die wordt uitgedrukt in de bewoordingen zoals 'geen GL/PvdA', 'stop die asielwaanzin' en 'luisteren naar die linkse flauwekul'. Deze bewoordingen geven een sterke afkeuring van linkse politieke ideeën en benadrukken de behoefte aan een rechts kabinet, wat duidt op een negatieve houding ten opzichte van linkse politieke partijen."), 
  set_train(variable = "political_stance_training", 
            value = 'Engiszins rechts/ conservatief -- Rechts/conservatief -- Helemaal rechts/ conservatief', 
            message = "Het politieke standpunt van deze tweet kan worden beschouwd als rechts of conservatief, vanwege de steun voor een rechts kabinet en de afwijzing van linkse partijen zoals GroenLinks (GL) en de Partij van de Arbeid (PvdA). De tweet drukt de wens uit voor beleid dat gericht is op het stoppen van wat wordt gezien als linkse maatregelen, zoals het beperken van asiel en het benadrukken van de behoefte aan meer woningen. Dit suggereert een voorkeur voor conservatieve beleidsmaatregelen en een afkeer van progressieve of linkse initiatieven."),
  set_train(variable = "misinformation_training", 
            value = 'Mogelijk/ Waarschijnlijk/ Zeker misinformatie', 
            message = "De tweet bevat onjuiste beweringen, zoals het beschuldigen van 'linkse flauwekul' en het gebruik van termen als 'asielwaanzin'. Deze uitspraken zijn subjectief en overdreven, en ze geven mogelijk een vertekend beeld van de realiteit. Daarom kan worden geconcludeerd dat de tweet misinformatie bevat."), 
  set_train(variable = "toxic_training", 
            value = "Enigzins toxisch -- Toxisch -- Zeer toxisch", 
            message = "Deze tweet bevat toxische taal omdat het negatieve bewoordingen gebruikt om een politiek standpunt over te brengen en andere politieke partijen bekritiseert op een respectloze manier"))
   

##=============

train_unit_2 <- create_unit('test vraag 2',
                          type = 'train',
                          set_text('text', 'Geweldig nieuws dat het gerechtshof heeft besloten dat de regering moet stoppen met de levering van F-35 onderdelen aan Israël. Eindelijk een stap in de richting van een rechtvaardiger buitenlands beleid dat de Palestijnse zaak serieus neemt. Dit geeft hoop voor een vreedzamere toekomst in het Midden-Oosten!'),
                          set_train(variable = "sentiment_training", 
                                    value = 'Enigszins positief -- Positief -- Zeer positief', 
                                    message = "Het sentiment van de tweet is positief omdat de auteur enthousiast reageert op het nieuws dat het gerechtshof heeft besloten om de levering van F-35 onderdelen aan Israël te stoppen. De auteur beschouwt dit als een stap in de richting van een rechtvaardiger buitenlands beleid en een serieuze erkenning van de Palestijnse zaak. De bewoordingen 'Geweldig nieuws' en 'Eindelijk een stap in de richting van een rechtvaardiger buitenlands beleid' geven aan dat de auteur het besluit toejuicht en hoopvol is over de mogelijke impact ervan op de toekomst van het Midden-Oosten."),
                          set_train(variable = "political_stance_training", 
                                    value = 'Volledig links/ progressief -- Links/ progressief -- Enigszins links/progressief', 
                                    message = 'Het politieke standpunt van de tweet kan worden beschouwd als links of progressief. De auteur ondersteunt het besluit van het gerechtshof om de levering van F-35 onderdelen aan Israël stop te zetten en beschouwt dit als een stap in de richting van een rechtvaardiger buitenlands beleid. Door te spreken over een "rechtvaardiger buitenlands beleid" en de "Palestijnse zaak serieus te nemen", lijkt de auteur te pleiten voor een meer evenwichtige en empathische benadering van de kwestie Israël-Palestina, wat over het algemeen wordt geassocieerd met progressieve politieke standpunten.'), 
                          set_train(variable = "misinformation_training", 
                                    value = 'Geen/ Waarschijnlijk geen/ Mogelijk geen misinformatie', 
                                    message = "De tweet bevat mogelijk geen misinformatie, aangezien het bericht gaat over een gerechtelijke beslissing en geen feitelijke onjuistheden lijkt te bevatten."),
                          set_train(variable = "toxic_training",
                                    value = 'Zeer beleefd/ vriendelijk -- Beleefd/ vriendelijk -- Enigzins beleefd/vriendelijk',
                                    message = "De tweet lijkt niet toxisch te zijn, aangezien de boodschap positief is en hoopvolle perspectieven biedt voor een vreedzamere toekomst in het Midden-Oosten."))
   
#===================

units <- c(intro_unit_1, intro_unit_2, train_unit_1,  train_unit_2)

codebook <- create_codebook(political_stance_training, sentiment_training, toxic_training, misinformation_training)

create_job('voorbeeld', units=units, codebook=codebook) |>
  create_job_db(overwrite=T) |>
  start_annotator()

### Part 2: Final coding job

<div style="background-color:yellow; padding:10px;">
@Kasper: General question about the codebook: Is it possible to display all 4 questions (political_stance, sentiment, toxic, misinformation) underneath each other, on one page?
</div>


In [42]:
## main codebook annotation

## misinformation
political_stance <- question("political_stance", 
                  "In welke mate reflecteert dit bericht een links- of rechts politiek standpunt?",
             instruction = read_and_concatenate("instructions/political_stance.txt")  , 
             codes = c( darkgrey = 'Volledig links/ progressief', 
                        grey = 'Links/ progressief', 
                        lightgrey= 'Enigszins links/progressief',
                        grey = 'Neutraal/ onduidelijk', 
                        lightred = 'Engiszins rechts/ conservatief',
                        grey = 'Rechts/conservatief',
                        darkgrey = 'Helemaal rechts/ conservatief'))

## sentiment
sentiment <- question("sentiment", 
                      "Is deze tweet negatief of positief van toon?",
             instruction = read_and_concatenate("instructions/sentiment.txt") , 
             codes = c( darkred = 'Zeer negatief',
                        red = 'Negatief',
                        lightred = 'Enigszins negatief',
                        grey = 'Neutral/onduidelijk', 
                        lightgreen = 'Enigszins positief',
                        green = 'Positief', 
                        darkgreen = 'Zeer positief'))

## toxic
toxic <- question("toxic", 
                  "Bevat deze tweet toxische taal?",
             instruction = read_and_concatenate("instructions/toxic.txt")  , 
             codes = c( darkgreen = 'Zeer beleefd/ vriendelijk', 
                        green = 'Beleefd/ vriendelijk', 
                        lightgreen = 'Enigzins beleefd/vriendelijk',
                        grey = 'Neutraal/ onduidelijk', 
                        lightred = 'Enigzins toxisch',
                        red = 'Toxisch',
                        darkred = 'Zeer toxisch'))

## misinformation
misinformation <- question("misinformation", 
                  "Bevat deze tweet misinformatie?",
             instruction = read_and_concatenate("instructions/misinformation.txt")  , 
             codes = c( darkgreen = 'Tweet bevat zeker geen misinformatie', 
                        green = 'Tweet bevat waarschijnlijk geen misinformatie', 
                        lightgreen = 'Tweet bevat mogelijk geen misinformatie',
                        grey = 'Neutraal/ onduidelijk', 
                        lightred = 'Bericht bevat mogelijk misinformatie',
                        red = 'Bericht bevat waarschijnlijk misinformatie',
                        darkred = 'Bericht bevat zeker misinformatie'))

### Create units for condition 1, 2, 3 (zonder Author):

<div style="background-color:yellow; padding:10px;">
@Kasper: Iedere codeur moet 30 tweets annoteren. In alle condities moet de hele dataset roteren; er zijn 3330 unique tweets, en ongeveer 330 participanten per conditie. We willen 3 unieke annotaties per tweet per conditie. 

    
Participanten komen van Qualtrics. Er zijn 6 verschillende redirect links vanuit Qualtrics mogelijk:

e.g., :

https://annotinder.com/?id={}?condition={}

So, something like: 

https://annotinder.com/?id=265179?condition=1  
https://annotinder.com/?id=265609?condition=2  
https://annotinder.com/?id=265902?condition=3  
https://annotinder.com/?id=265632?condition=4  
https://annotinder.com/?id=303467?condition=5  
https://annotinder.com/?id=358590?condition=6  
    

conditie 1 t/ 3 moet deze units te zien krijgen: 

```r
units_condition_1_2_3 <- create_units(df,            
  set_text("text", text, align = "left", text_size = 1.5, bold = TRUE, underline = TRUE),
  id = "id"
)
```
                         
conditie 3 t/ 6 moet deze units te zien krijgen: 
   
```r
units_condition_3_4_5 <- create_units(df,
  set_markdown("review_message", review_message, align="left", text_size = 1),                 
  set_text("text", text, align = "left", text_size = 1.5, bold = TRUE, underline = TRUE),
  id = "id",
  set_markdown("author_with_handle", author_with_handle, align = "left", text_size = 1.2)
)
```
    
Ik kan de basis url uiteraard aanpassen; laat maar weten waar participanten heen moeten!

</div>


In [45]:
## Create units for final experiment... 
units_condition_1_2_3 <- create_units(df,
  #set_markdown("review_message", review_message, align="left", text_size = 1),                 
  set_text("text", text, align = "left", text_size = 1.5, bold = TRUE, underline = TRUE),
  id = "id"
  #set_markdown("author_with_handle", author_with_handle, align = "left", text_size = 1.2)
)

codebook <- create_codebook(sentiment, toxic, misinformation, political_stance)
job <- create_job("simple_example", units_condition_1_2_3 , codebook)
job_db <- create_job_db(job, overwrite = T)
job_db

start_annotator(job_db)

[1] "/Users/anne/repos/AnnoBias/AnnoTinder/annotinder_jobs/simple_example.db"

### Create units for conditions 3, 4, 5 (met Author): 

In [27]:
units_condition_3_4_5 <- create_units(df,
  set_markdown("review_message", review_message, align="left", text_size = 1),                 
  set_text("text", text, align = "left", text_size = 1.5, bold = TRUE, underline = TRUE),
  id = "id",
  set_markdown("author_with_handle", author_with_handle, align = "left", text_size = 1.2)
)

In [44]:
codebook <- create_codebook(sentiment, toxic, misinformation, political_stance)
job <- create_job("simple_example", units_condition_1_2_3 , codebook)
job_db <- create_job_db(job, overwrite = T)
job_db

start_annotator(job_db)

[1] "/Users/anne/repos/AnnoBias/AnnoTinder/annotinder_jobs/simple_example.db"

In [12]:
start_annotator(job_db)